Source/Core/Core/State.cpp

```diff
@@ -220,14 +220,18 @@ static void DoState(PointerWrap& p)
   g_video_backend->DoState(p);
   p.DoMarker("video_backend");
 
-  PowerPC::DoState(p);
-  p.DoMarker("PowerPC");
   // CoreTiming needs to be restored before restoring Hardware because
   // the controller code might need to schedule an event if the controller has changed.
   CoreTiming::DoState(p);
   p.DoMarker("CoreTiming");
+
+  // HW needs to be restored before PowerPC because the data cache might need to be flushed.
```

Comment:
You should probably also update [the `STATE_VERSION` over here](https://github.com/Lobsterzelda/lobster-zelda-dolphin/blob/master/Source/Core/Core/State.cpp#L97) if you change the order.